src/types.rs

```diff
@@ -61,6 +61,10 @@ impl Operator {
     }
 }
 
+/// ['Atom'] enum which holds either a ['Variable'] or an ['Operator']
```

Comment:
This is a good point. The key thing about an `Atom`, and the reason we call it an `Atom`, is because it represents an indivisible part, the smallest things of which `Term`s are made: the `Operator` and the `Variable`. These are the building blocks of `Term`s.

Perhaps a more general point to notice here is that docs shouldn't typically say what the code already says. They should summarize and provide semantic context. Why is something the way it is? What do you expect people to do with it? etc.